# iPSCs_VariantAnalysis
Methodology for analyzing pathogenic variants within the genome of patient-derived induced pluripotent stem cells

## Analysis Workflow
This analysis pipeline is based on the GATK Best Practices and ACMG Guidelines, and aims to define the necessary content for extracting and evaluating risk variants from the whole exome or whole genome sequence of patient-derived induced pluripotent stem cells (iPSCs)


본 분석파이프라인은 GATK Best Practices 및 ACMG Guidelines 에 기반하여 수행되었으며, 환자 유래의 유도만능줄기세포(iPSC)의 전장 엑솜 혹은 전장 유전체 염기서열로부터 위험 변이를 추출하고 평가하는데에 필요한 내용을 규정하고자 하는 목적이 있음


### Requirements
*  Hg38 reference (resources_broad_hg38_v0_Homo_sapiens_assembly38.fasta)
*  Bwa (version: 0.7.17-r1188)
*  Samtools (version: 1.13, using htslib 1.13+ds)
*  GATK4
*  InterVar
*  ANNOVAR
*  NGSCheckMate

### Variant Calling
The Genome Analysis Toolkit (GATK) Best Practices for variant calling on whole exome or whole genome sequencing data involve several key steps.
* Pre-processing of Raw Sequencing Data :
    * Quality Control : Assess the quality of the raw sequencing data.
    * Alignment : Align the reads to a reference genome using BWA (Burrows-Wheeler Aligner).
    * Marking Duplicates : Identify and mark duplicate reads to avoid counting the same DNA fragment more than once.
* Post-alignment Processing :
    * Base Quality Score Recalibration (BQSR) : Adjust the base quality scores to correct for systematic errors made by the sequencer.
    * Indel Realignment : Realign reads around indels (insertions and deletions) to correct alignment errors.
    * Base Quality Score Recalibration (again) : After indel realignment, perform another round of BQSR.
* Variant Discovery :
    * Variant Calling : Use tools like HaplotypeCaller (for single samples) or Mutect2 (for fibroblast/iPSC pairs) to identify variants (SNPs and indels).
    * Joint Genotyping : If analyzing multiple samples, combine individual GVCFs (generated by HaplotypeCaller) into a single VCF file. (Optional)
* Variant Filtration :
    * Hard-filtering : Filter variant calls based on INFO and/or FORMAT annotations.


### QC for validation sample identity of sequencing data
*  To ensure that iPSC lines from each individual are properly paired
*  Uses correlation models of allele fractions of known SNVs to identify samples from same individual (PNAS, 2017)

### Functional Annotation and Interpretation:
* Annotation : Use tools like Annovar or VEP (Variant Effect Predictor) to annotate variants based on their potential impact on genes.
* Interpretation : Assess the clinical significance of variants, guided by databases like ClinVar.

### iPSC line selection based on genetic variations and their potential clinical significance

  1. Genetic variations in neurodegenerative disorder, cancer related genes (n=965)
  2. How genetic variants affect transcript structure or coding sequences
  3. Coding variants - deleterious to protein function or structure
  4. Functionally or clinically validated genetic variants


For more information, also see :


*  Bwa : https://bio-bwa.sourceforge.net/
*  Samtools : http://www.htslib.org/
*  GATK : https://gatk.broadinstitute.org/hc/en-us/sections/360007226651-Best-Practices-Workflows
*  ACMG Guidelines : https://www.acmg.net/docs/standards_guidelines_for_the_interpretation_of_sequence_variants.pdf
*  InterVar : https://wintervar.wglab.org/
*  ANNOVAR : https://annovar.openbioinformatics.org/en/latest/
*  NGSCheckMate : https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5499645/pdf/gkx193.pdf



